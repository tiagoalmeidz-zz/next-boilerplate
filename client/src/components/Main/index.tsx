const Main = () => (
  <main>
    <h1>Boilerplate Project</h1>
  </main>
)

export default Main
