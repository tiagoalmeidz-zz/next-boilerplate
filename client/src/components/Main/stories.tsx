import { withKnobs, text } from '@storybook/addon-knobs'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
}

export const Basic = () => (
  <Main
    title={text('Title', 'Boilerplate Project')}
    description={text(
      'Description',
      'TypeScript, ReactJS, NextJS and Styled Components'
    )}
  />
)
