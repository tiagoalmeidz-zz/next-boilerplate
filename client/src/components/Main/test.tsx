import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the headings', () => {
    render(<Main />)

    expect(screen.getByRole('heading', { name: /boilerplate project/i }))
  })
})
