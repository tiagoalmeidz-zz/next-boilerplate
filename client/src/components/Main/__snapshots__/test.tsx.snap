// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the headings 1`] = `
<main
  class="styles__Wrapper-sc-1cs6c8-0 bBVefU"
>
  <img
    alt="Imagem da logo do webpack."
    class="styles__Logo-sc-1cs6c8-1 eivRWX"
    src="/img/webpack-512.png"
  />
  <h1
    class="styles__Title-sc-1cs6c8-2 bQPDep"
  >
    Boilerplate Project
  </h1>
  <h2
    class="styles__Description-sc-1cs6c8-3 jApXu"
  >
    TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
    class="styles__Illustration-sc-1cs6c8-4 cZHMdn"
    src="/img/hero-illustration.svg"
  />
</main>
`;
